### DISCUSSION TITLE
Font loading before rendering

### DISCUSSION THREAD

**Discussion by wusikijeronii** [ocornut/imgui#7289] [New Users Build/Link/Run issues ONLY!]
Hello,
I cache some data during the resources preparation stage to reduce the number of calls per frame. Today I ran into an error trying to get the text size. To determine the size, I use the `ImGui::CalcTextSize` function. But as I found out, it uses font data (which is logical), but this font itself is not loaded at the initialization stage (through the `AddFontFromFileTTF` function), and it is NULL. Initialization occurs in the first frame. Is there a way to load the font before the first call to `ImGui::New Frame`? I will add that for non-GUI tasks, I use my implementation of text atlases. And I prepare the data in advance. So I don't see a problem with that.

**Comment by PathogenDavid** [comment]
Yes and no. You can build the font atlas and associated data before `ImGui::NewFrame` using `ImFontAtlas::Build`, but `ImGui::CalcTextSize` still won't work as it relies on other state in `ImGuiContext` to be set up by `NewFrame`.

You can however use `ImFont::CalcTextSizeA` instead, which is slightly different from `ImGui::CalcTextSize` but not in ways that'd likely affect your use-case.

Here's an example:

```cpp
ImGuiIO& io = ImGui::GetIO();
io.Fonts->Build();
ImFont* font = io.Fonts->Fonts[0];
ImVec2 size = font->CalcTextSizeA(font->FontSize, FLT_MAX, 0.f, "Hello, world!");
```

**Reply by wusikijeronii** [reply]
Yes, it can work. Thanks, I just had to ask. In case there was a direct way to do this.
